integration-tests/tests/src/tests/sync/user.ts

```diff
@@ -102,43 +106,60 @@ describe.skipIf(environment.missingServer, "User", () => {
       const provider = this.app.emailPasswordAuth;
       expect(provider).instanceof(Realm.Auth.EmailPasswordAuth);
     });
+  });
 
-    it("autoverify email password works", async function (this: AppContext & RealmContext) {
-      const validEmail = randomVerifiableEmail();
-      const invalidEmail = randomNonVerifiableEmail();
-      const invalidPassword = "pass"; // too short
-      const validPassword = "password123456";
+  describe("autoverify email password works", () => {
+    importAppBefore(buildAppConfig("with-email-password").emailPasswordAuth());
+    removeExistingUsers();
+
+    const validEmail = randomVerifiableEmail();
+    const invalidEmail = randomNonVerifiableEmail();
+    const invalidPassword = "pass"; // too short
+    const validPassword = "password123456";
 
-      // invalid email, invalid password
-      let credentials = Realm.Credentials.emailPassword({ email: invalidEmail, password: invalidPassword });
+    it("invalid email, invalid password", async function (this: AppContext & RealmContext) {
+      const credentials = Realm.Credentials.emailPassword({ email: invalidEmail, password: invalidPassword });
       await expect(this.app.logIn(credentials)).to.be.rejectedWith("invalid username/password"); // this user does not exist yet
       await expect(
         this.app.emailPasswordAuth.registerUser({ email: invalidEmail, password: invalidPassword }),
-      ).to.be.rejectedWith("password must be between 6 and 128 characters");
+      ).to.eventually.be.rejectedWith("password must be between 6 and 128 characters");
       await expect(this.app.logIn(credentials)).to.be.rejectedWith("invalid username/password"); // this user did not register
+    });
 
-      // invalid email, valid password
-      credentials = Realm.Credentials.emailPassword({ email: invalidEmail, password: validPassword });
+    // any email address is valid and registerUser() will never fail -- the test doesn't make sense
+    it.skip("invalid email, valid password", async function (this: AppContext & RealmContext) {
+      const credentials = Realm.Credentials.emailPassword({ email: invalidEmail, password: validPassword });
       await expect(this.app.logIn(credentials)).to.be.rejectedWith("invalid username/password"); // this user does not exist yet
       expect(
         this.app.emailPasswordAuth.registerUser({ email: invalidEmail, password: validPassword }),
-      ).to.be.rejectedWith(`failed to confirm user "${invalidEmail}"`);
+      ).to.eventually.be.rejectedWith(`failed to confirm user "${invalidEmail}"`);
       await expect(this.app.logIn(credentials)).to.be.rejectedWith("invalid username/password"); // this user did not register
+    });
```

Comment:
Looks like this test belongs in a "custom function" test. I agree with the comment that for an auto confirm app, the login should not be rejected.

Reply:
We should remove the test instead of skipping it

Reply:
Yeah, you could probably delete that entire `it` test. Additionally, perhaps after the `"custom confirmation function works"` you could add the following (which is pretty much the same test):

```ts
it("custom confirmation function rejects invalid email", async function (this: AppContext & RealmContext) {
      const invalidEmail = randomNonVerifiableEmail();
      const validPassword = "password123456";

      const credentials = Realm.Credentials.emailPassword({ email: invalidEmail, password: validPassword });
      await expect(this.app.logIn(credentials)).to.be.rejectedWith("invalid username/password"); // this user does not exist yet
      expect(
        this.app.emailPasswordAuth.registerUser({ email: invalidEmail, password: validPassword }),
      ).to.eventually.be.rejectedWith(`failed to confirm user "${invalidEmail}"`);
      await expect(this.app.logIn(credentials)).to.be.rejectedWith("invalid username/password"); // this user did not register
    });
```

I know that `describe` suite is currently skipped, but at least we still keep the test around.